src/services/fields.ts

```diff
@@ -53,8 +55,23 @@ export function extractParserAndFieldsFromDataFrame(data: DataFrame) {
     }, {}) ?? {}
   );
 
+  const types: ExtractedFieldsType[] = [];
+
   const linesField = data.fields.find((f) => f.name === 'Line' || f.name === 'body');
-  result.type = linesField?.values[0]?.[0] === '{' ? 'json' : 'logfmt';
+  linesField?.values.forEach((value: string) => {
+    if (types.length === 2) {
+      return;
+    }
+    try {
+      if (JSON.parse(value) && !types.includes('json')) {
```

Comment:
1000 logs:

- JSON.parse 8ms
- trim().startsWith({) 0.3ms

Reply:
We can check the first (and last) char before attempting the parse, but that only helps for mixed mode, or users with logfmt. 100% json would see the worse performance, and huge JSON lines will take longer to parse

Reply:
I don't think parsing is required, we can just check the start and end chars (great idea). Main goal: not just check [0][0]

Reply:
Changed the approach and removed parsing. Tested it with actual mixed logs and works as expected.

Perf: 1000 logs 0.10000002384185791 ms

Reply:
and that won't change (enough to matter) as the size of the json increases, opposed to parsing